File: fams_08042024_2307/fams-main/DataLayer/Entities/UserPermission.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DataLayer.Entities
{
    public class UserPermission
    {
        [Key]
        public int PermissionId { get; set; }
        [Required(ErrorMessage = "Name of permission is required")]
        [StringLength(100, ErrorMessage = "Name of permission is not longer than 100 characters.")]
        public string PermissionName { get; set; }

        //public Guid UserId { get; set; }

        public Permissions Syllabus { get; set; }
        public Permissions LearningMaterial { get; set; }
        public Permissions TrainingProgram { get; set; }
        public Permissions Class { get; set; }
        public Permissions UserManagement { get; set; }
        public int Version { get; set; }

        public virtual ICollection<User>? Users { get; set; }
    }
    public enum Permissions
    {
        AccessDenied,
        View,
        Modify,
        Create,
        FullAccess
    }
}

File: src/lib/schemas/user-permission.ts
import * as z from "zod"

import {
  IUserPermission,
  IUserRoleName
} from "@/types/user-permission.interface"
import { IUserRole } from "@/types/user.interface"

export const UserRoleConfig = z.object({
  permissionId: z.nativeEnum(IUserRole),
  permissionName: z.nativeEnum(IUserRoleName),
  syllabus: z.nativeEnum(IUserPermission),
  learningMaterial: z.nativeEnum(IUserPermission),
  trainingProgram: z.nativeEnum(IUserPermission),
  class: z.nativeEnum(IUserPermission),
  userManagement: z.nativeEnum(IUserPermission),
  version: z.nativeEnum(IUserPermission),
  users: z.any()
})

export const UserRoleConfigSchema = z.array(UserRoleConfig)
